newton/_src/geometry/broad_phase_sap.py

```
@@ -345,94 +440,79 @@ def _sap_broadphase_kernel(
                 max_candidate_pair,
+            )
+            workid += nsweep_in
+        return
+
+    # Reuse interval lookups only when the original threads would loop.
+    chunk_size = 4
+    chunk_base = tid * chunk_size
+    chunk_stride = nsweep_in * chunk_size
+    while chunk_base < total_work_packages:
+        flat_id = binary_search(sap_cumulative_sum_in, chunk_base, 0, total_intervals)
+        chunk_offset = int(0)
```

Ruff (0.16.1)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)